src/main/java/net/hypercubemc/universe_installer/UniverseConfig.java

```diff
@@ -0,0 +1,97 @@
+package net.hypercubemc.universe_installer;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class UniverseConfig {
+    private String selectedEditionName;
+    private String selectedEditionDisplayName;
+    private String selectedVersion;
+    private Path customInstallDir;
+    private boolean useCustomLoader;
+
+    private final Path propertiesPath;
+    private final Properties properties;
+
+
+    public UniverseConfig() {
+        selectedEditionName = null;
+        selectedEditionDisplayName = null;
+        selectedVersion = null;
+        customInstallDir = null;
+        useCustomLoader = true;
+
+        Installer installer = new Installer();
+        propertiesPath = installer.getStorageDirectory().resolve("universe.properties");
+        properties = new Properties();
+    }
+
+    public boolean getUseCustomLoader() {
+        return useCustomLoader;
+    }
+
+    public void setUseCustomLoader(boolean useCustomLoader) {
+        this.useCustomLoader = useCustomLoader;
+    }
+
+    public String getSelectedEditionName() {
+        return selectedEditionName;
+    }
+
+    public String getSelectedEditionDisplayName() {
+        return selectedEditionDisplayName;
+    }
+
+    public String getSelectedVersion() {
+        return selectedVersion;
+    }
+
+    public Path getCustomInstallDir() {
+        return customInstallDir;
+    }
+
+    public void setSelectedEditionName(String s) {
+        selectedEditionName = s;
+    }
+
+    public void setSelectedEditionDisplayName(String s) {
+        selectedEditionDisplayName = s;
+    }
+
+    public void setCustomInstallDir(Path p) {
```

Comment:
Same type of problem as the previous comments, rename the argument to `path` and change the setter to use `this`